### ISSUE TITLE
Can't use fieldsForIndex / fieldsForDetail / fieldsForCreate / fieldsForUpdate

### ISSUE THREAD

**Issue by jp-gauthier** [epartment/nova-dependency-container#159]
Hi, when I use this package, I can't use functions like `fieldsForIndex / fieldsForDetail / fieldsForCreate / fieldsForUpdate` in my Nova resource.

The method `HasDependencies.php : availableFields` seems to override the Nova `ResolvesFields.php : availableFields`.

When I comment your `availableFields` method, it works normally.

Please correct or suggest a workaround. Thanks.

**Comment by ragingdave** [comment]
The workaround would likely just be to try and use different dependency containers for each, and restrict those. I haven't tested that specifically but that could work. Also feel free to submit a PR with a solution if you have one that works well.

**Comment by jp-gauthier** [comment]
@ragingdave Hello, hmm I'm not sure how to do that. Is it possible without changing the code in your package ?

**Comment by ragingdave** [comment]
Yes. Not a great workaround, but quite literally duplicate the dependency Container  and add on the nova methods to show them specifically when you want them. https://nova.laravel.com/docs/3.0/resources/fields.html#showing-hiding-fields This would be added to only the container field itself, and not the array of fields passed.

It's basically redefining the fields multiple times, but the visibility would be applied correctly based on how you break things out.

**Comment by jp-gauthier** [comment]
Ok nice, I'll test it out !

**Comment by anditsung** [comment]
i try to update HasDependencies and it was working fine on mine.

what i change

```
$fields = $this->filter($this->fields($request));
```

to
```
$method = $this->fieldsMethod($request);
$fields = $this->filter($this->{$method}($request));
```

**Comment by jp-gauthier** [comment]
@anditsung Nice, it works! I just saw someone made a pull request yesterday with this fix : #170 

@ragingdave Can you merge the pull request so it can be included in the next version? Thanks

**Comment by BinaryKitten** [comment]
Thanks Team,

The PR #170 was merged and as of today released as [v1.3.3](https://github.com/epartment/nova-dependency-container/releases/tag/1.3.3) as noted on #169
